### DISCUSSION TITLE
Does lazy loading break ":tab ..."?

### DISCUSSION THREAD

**Discussion by Hubro** [wbthomason/packer.nvim#622] [Q&A]
I'm lazy-loading Fugitive like this:

```lua
  -- {{{ fugitive.vim - Premier Git plugin for Vim
  use {
    "tpope/vim-fugitive",
    opt = true,
    cmd = {
      "G", "Git", "Gdiffsplit", "Gvdiffsplit", "Gedit", "Gsplit",
      "Gread", "Gwrite", "Ggrep", "Glgrep", "Gmove",
      "Gdelete", "Gremove", "Gbrowse",
    },
  }
  -- }}}
```

When I do this, the first time I call `:tab G`, Fugitive is opened in a regular split, just as if I had called `:G`.

The next time I call `:tab G`, Fugitive opens in a new tab.

I assume this is because packer creates these commands, and when they are called, it deletes the commands, loads the right plugin and then calls the command again. Could this be the case? If so, is it possible for packer to detect the use of `:tab ...` and forward it to the command after the plugin has been loaded?

**Comment by Hubro** [comment]
Aha, I see. The proxy-command calls the real command asynchronously via `au CmdUndefined`, so prefixing the proxy-command with `:tab` has no effect. To fix this, the proxy-command has to "detect" that `:tab` is being used somehow and conditionally prefix the real command with it as well.

**Reply by wbthomason** [reply]
Noting for posterity that this is fixed by #624 and #642.
